ros/CMakeLists.txt

```diff
@@ -65,18 +65,32 @@ elseif("$ENV{ROS_VERSION}" STREQUAL "2")
 
   find_package(ament_cmake REQUIRED)
   find_package(nav_msgs REQUIRED)
+  find_package(rcutils REQUIRED)
   find_package(rclcpp REQUIRED)
+  find_package(rclcpp_components REQUIRED)
   find_package(sensor_msgs REQUIRED)
   find_package(tf2_ros REQUIRED)
 
   # ROS2 node
-  add_executable(odometry_node ros2/OdometryServer.cpp)
```

Comment:
I'm new to this. This "component" can also be launched as a "normal node"? Is this backward compatible with the existing launch files?

Reply:
I'll leave some details here :)

Basically, your class becomes a component, which can be dynamically loaded by a ComponentManager. So that's why I renamed the target and changed from `add_executable` to `add_library`, and also removed the `main()`. Then, `rclcpp_components_register_node` macro in CMake generates a second target (using a default `main` routine) called like the old node. This target is, like the node you were building prior to this patch, a standalone executable that behaves exactly like a node. The only thing is, now it is dynamically linked to the component target (no user will notice this).

Reply:
So yes, it shall be compatible with existing launchfiles (if you can, have a try).